tracer.go

```diff
@@ -52,38 +76,54 @@ func (t *Tracer) StartSpan(
 		option(t, s)
 	}
 
+	if (model.SpanContext{}) != s.SpanContext {
```

Comment:
Can we ask a SpanContext whether it is a noop or not instead of this? something like `s.SpanContext.IsNoop()`.

Reply:
and move this exact same construct to test for zero value into a SpanContext method? Don't forget this is tracer internal code, no consumer needs to touch. We just lose efficiency with 0 gain...

Reply:
For the moment we rely an empty context is a `zero value` context but at some point could this change?

Reply:
I don't see that happen any time soon. If it does it will not only affect the question of IsNoop() but also means adjusting the very code where the test is currently being done... So I'll leave as is because of YAGNI